ext/gmp/gmp.c

```diff
@@ -609,16 +619,34 @@ static int convert_to_gmp(mpz_t gmpnumber, zval *val, zend_long base)
 
 		ret = mpz_set_str(gmpnumber, (skip_lead ? &numstr[2] : numstr), (int) base);
 		if (-1 == ret) {
-			php_error_docref(NULL, E_WARNING,
-				"Unable to convert variable to GMP - string is not an integer");
+			/* Try to guess base for better error message */
+			if (base == 0) {
+				if (numstr[0] == '0' || ((numstr[0] == '+' || numstr[0] == '-') && numstr[1] == '0')) {
+					base = 8;
+				} else { /* Default to base 10 */
+					base = 10;
+				}
+			}
+
+			/* if unserializing */
+			if (arg_pos == 0) {
+				php_error_docref(NULL, E_WARNING,
+					"Cannot convert variable to GMP, string is not an integer in base " ZEND_LONG_FMT, base);
+				return FAILURE;
+			}
+			zend_argument_type_error(arg_pos, "must be an integer string in base " ZEND_LONG_FMT, base);
 			return FAILURE;
 		}
 
 		return SUCCESS;
 	}
 	default:
-		php_error_docref(NULL, E_WARNING,
-			"Unable to convert variable to GMP - wrong type");
+		/* if unserializing */
+		if (arg_pos == 0) {
+			php_error_docref(NULL, E_WARNING, "Cannot convert variable of type %s to GMP", zend_zval_type_name(val));
+			return FAILURE;
+		}
+		zend_argument_type_error(arg_pos, "must be of type bool|int|string|GMP, %s given", zend_zval_type_name(val));
```

Comment:
The "standard" order of the types are `GMP|string|int|bool`, so maybe it's worth to follow this format here as well.